middleware/error-handler.js

```diff
@@ -1,10 +1,37 @@
-const { CustomAPIError } = require('../errors')
-const { StatusCodes } = require('http-status-codes')
+// const { CustomAPIError } = require('../errors');
+const { StatusCodes } = require('http-status-codes');
 const errorHandlerMiddleware = (err, req, res, next) => {
-  if (err instanceof CustomAPIError) {
-    return res.status(err.statusCode).json({ msg: err.message })
+  let customError = {
+    //set default
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something went wrong try again later',
```

Comment:
We should only send `err.message` back to the client if it's an error type that we expect. In other words, if it's specifically an instance of CustomAPIError.

The base `Error` type in javascript has a `.message` property, so the error message that you're sending back could be from any place whatsoever!

In general, when an unexpected error occurs, exposing those details to the outside web is a bad practice – it gives a potential attacker more details about what is going wrong, which we want to avoid.

To fix this, make sure that you're not sending any not-null `err.message`, but that you're _only_ sending `err.message` if it is one of your own CustomAPIErrors – otherwise, you don't know where that thing came from, and you might not want to tell attackers what's going on behind the scenes!